eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1166,15 +1166,15 @@ class Channel(val nodeParams: NodeParams, wallet: EclairWallet, remoteNodeId: Pu
     case Event(INPUT_RECONNECTED(r), d: HasCommitments) =>
       forwarder ! r
 
-      val yourLastPerCommitmentSecret = d.commitments.remotePerCommitmentSecrets.lastIndex.flatMap(d.commitments.remotePerCommitmentSecrets.getHash).map(Scalar(_))
-      val myCurrentPerCommitmentPoint = Some(Generators.perCommitPoint(d.commitments.localParams.shaSeed, d.commitments.localCommit.index))
+      val yourLastPerCommitmentSecret = d.commitments.remotePerCommitmentSecrets.lastIndex.flatMap(d.commitments.remotePerCommitmentSecrets.getHash).getOrElse(Sphinx zeroes 32)
```

Comment:
There is a complication with using `None` here. 

Even if it's our first commit we still need to send `myCurrentPerCommitmentPoint` so the message can't be `ChannelReestablish(..., None, None)` and setting it to `ChannelReestablish(..., None, Some(myCurrentPerCommitmentPoint))` causes an exception when peer tries to deserialize it, presumably because `Some(myCurrentPerCommitmentPoint)` falls into `yourLastPerCommitmentSecret`'s place since `yourLastPerCommitmentSecret` is `None`.

I've got back to sending zeroes for now, is there a better solution?

Reply:
Ok, I've submitted PR https://github.com/lightningnetwork/lightning-rfc/pull/390 to clarify in the specs.